widget/button.go

```diff
@@ -86,7 +91,11 @@ func (b *buttonRenderer) Refresh() {
 			b.icon = canvas.NewImageFromResource(b.button.Icon)
 			b.objects = append(b.objects, b.icon)
 		} else {
-			b.icon.Resource = b.button.Icon
+			if b.button.disabled {
+				b.icon.Resource = b.button.disabledIcon
```

Comment:
There is a bug somewhere here.
The button.Icon can change - but the disabledIcon will not change to match it.
Here when we return the image it may be out of date...

Reply:
Great catch!  So, the most efficient solution (that I've come up with) is to add some logic to the `button.SetIcon()` function to also update the `button.disabledIcon`.  However, it looks like `button.Icon` is public and could be mutated directly.  To handle situations like this I've added another level of conditionals to the `button.Refresh()` function.  Let me know what you think.  Pushed.